app/models/mixins/relationship_mixin.rb

```diff
@@ -580,9 +580,13 @@ def child_types(*args)
   end
 
   def add_parent(parent)
-    return update(:parent => parent) if use_ancestry?
-
-    parent.with_relationship_type(relationship_type) { parent.add_child(self) }
+    parent.with_relationship_type(relationship_type) do
```

Comment:
I guess I would've expected that there would also need to be a `parent=` change.

Reply:
this is putting the onus on the caller to set the correct `relationship_type` on `parent`